agr_java_core/src/main/java/org/alliancegenome/neo4j/repository/AlleleRepository.java

```diff
@@ -439,25 +436,50 @@ private String getCypherQuery(String relationship) {
         return allAllelicHgvsGNames;
     }
     public Map<String, Allele> getAllAlleleVariants() {
-
+//{primaryKey:'RGD:728326'}
         String query = "MATCH p1=(:Species)<-[:FROM_SPECIES]-(a:Allele)";
         query += " OPTIONAL MATCH gene=(a:Allele)-[:IS_ALLELE_OF]-(g:Gene) ";
         query += " OPTIONAL MATCH p=(a:Allele)<-[:VARIATION]-(variant:Variant)-[:ASSOCIATION]->(:TranscriptLevelConsequence)--(:Transcript)--(:SOTerm)";
         query += " OPTIONAL MATCH crossRef=(a:Allele)-[:CROSS_REFERENCE]->(c:CrossReference)";
         query += " OPTIONAL MATCH vari=(a:Allele)<-[:VARIATION]-(variant:Variant)-[:VARIATION_TYPE]->(soTerm:SOTerm)";
         query += " OPTIONAL MATCH p2=(a:Allele)-[:ALSO_KNOWN_AS]->(synonym:Synonym)";
-        query += " RETURN  p,crossRef,vari,p2,p1, gene";
+        query += " OPTIONAL MATCH consequence=(variant:Variant)-[:ASSOCIATION]->(:GeneLevelConsequence)";
+        query += " OPTIONAL MATCH loc=(variant:Variant)-[:ASSOCIATION]->(:GenomicLocation)-[:ASSOCIATION]->(:Chromosome)";
+
+        query += " RETURN  p,crossRef,vari,p2,p1, gene, consequence,loc";
 
         Iterable<Allele> allelesWithVariantsIter = query(query, new HashMap<>());
         Set<Allele> allelesWithVariants = StreamSupport.stream(allelesWithVariantsIter.spliterator(), false)
                 .collect(Collectors.toSet());
         Map<String, Allele> alleleVariantsMap=new HashMap<>();
         for(Allele a: allelesWithVariants){
+            if(a.getVariants()!=null && a.getVariants().size()>0){
+                for(Variant v:a.getVariants()){
+                    if(v.getTranscriptLevelConsequence()!=null && v.getTranscriptLevelConsequence().size()>0){
+                        for(TranscriptLevelConsequence c:v.getTranscriptLevelConsequence()){
+                            Transcript t=c.getTranscript();
+                            System.out.println(t.getPrimaryKey());
+                            Transcript transcript=getTranscriptWithExonInfo().get(t.getPrimaryKey());
+                            if(transcript.getExons()!=null){
+                                t.setExons(transcript.getExons());
+                               // System.out.println("EXONS SIZE: "+ transcript.getExons().size());
+                            }
+
+                        }
+                    }
+                }
+            }
             alleleVariantsMap.put(a.getPrimaryKey(), a);
         }
         log.info("Number of alleles with variants: " + String.format("%,d", allelesWithVariants.size()));
 
         return alleleVariantsMap;
     }
+    public static void main(String[] args){
+        AlleleRepository repo=new AlleleRepository();
+        repo.getAllAlleleVariants();
+        repo.close();
+        System.out.println("DONE!!");
+    }
```

Comment:
We typically put these kind of tests in the test package so that there are not main methods throughout the code.

Reply:
yes, removed it